### ISSUE TITLE
default measures enforce record syntax

### ISSUE THREAD

**Issue by zilinc** [ucsd-progsys/liquidhaskell#871]
In LH,
```
data Vec [vlen] t = Nil | Cons {td :: t, tl :: Vec t}  -- OK
data Vec [vlen] t = Nil | Cons t (Vec t)  -- Fails with awkward error message
```

**Comment by nikivazou** [comment]
Indeed annoying... Thanks for issuing!

**Comment by kosmikus** [comment]
Fixed via #1704.
